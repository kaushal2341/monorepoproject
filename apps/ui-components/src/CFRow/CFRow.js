import React from 'react';
import Form from "react-bootstrap/Form";
import PropTypes from 'prop-types';

const CFRow = props => {
    const {
        as,
        bsPrefix,
        children,
        className,
        id
    } = props;
    return (
        <Form.Row
            as={as}
            bsPrefix={bsPrefix}
            children={children}
            className={className}
            id={"fRow_".concat(id)}
        />
    )
};

CFRow.propTypes = {
    as: PropTypes.elementType,
    bsPrefix: PropTypes.string,
    children: PropTypes.node,
    id: PropTypes.string.isRequired
};

export default CFRow;