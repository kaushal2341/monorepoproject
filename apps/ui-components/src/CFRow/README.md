# CFRow 

Row component made using react-bootstrap's `Form.Row` component.
To use it ,first import the component,

```jsx harmony
import {CFRow} from '@cogent/ui-components';
```
Then use it as,

```jsx harmony
<CFRow id="email">
    <CFLabel
       id="name"
       labelName="Email"
       column={true}
       srOnly={false}
       md={4}
    />
    <CFControl
       id="email"
       type="email"
       name="email"
       placeholder="Enter your email"
       size="lg"
       required={true}
       onChange={(e) => console.log(e.target.value)}
    />
</CFRow>
```
## CFRow Props

```
as: elementType - defaultValue: <div> - You can use a custom element type for this component.

bsPrefix string - defaultValue: 'form-row' - Change the underlying component CSS base class name and modifier class names prefix. This is an escape hatch for working with heavily customized bootstrap css.

children: node - Children components.

className: string - Css classname for additional styling.

id: string - Id of the component.
```