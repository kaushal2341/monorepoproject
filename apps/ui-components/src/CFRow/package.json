{
  "name": "@cogent/c-f-row",
  "version": "0.0.0",
  "main": "CFRow.js",
  "author": "Sabu Shakya",
  "license": "MIT",
  "private": true
}
