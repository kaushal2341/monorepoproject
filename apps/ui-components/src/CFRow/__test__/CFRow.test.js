import CFRow from "../CFRow";

expect.addSnapshotSerializer(enzymeSerializer);

describe('CFRow Component Tests', () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CFRow id="test"/>);
    });

    it('should be defined', () => {
        expect(wrapper.find('CFRow')).toBeDefined();
    });

    it('should have only one Form.Row component', () => {
        expect(wrapper.find('#fRow_test').length).toBe(1);
    });

    it('should have all props available', () => {
        let propsAvailable = [
            'as',
            'bsPrefix',
            'children',
            'className',
            'id'
        ];
        let propsOfCrow = Object.keys(wrapper.find('#fRow_test').props());
        propsAvailable.forEach((propAvail, i) => (
            expect(propAvail).toContain(propsOfCrow[i])
        ));
    });

    describe('Snapshot Testing', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
    });
});