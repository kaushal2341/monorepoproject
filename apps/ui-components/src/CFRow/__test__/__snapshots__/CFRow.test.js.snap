// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFRow Component Tests Snapshot Testing should match snapshot 1`] = `
<FormRow
  id="fRow_test"
/>
`;
